watertap/costing/unit_models/ion_exchange.py

```diff
@@ -34,7 +34,7 @@ def build_hcl_cost_param_block(blk):
     )
 
     costing = blk.parent_block()
-    costing.add_defined_flow("HCl", blk.cost / blk.purity)
+    costing.register_flow_type("HCl", blk.cost / blk.purity)
```

Comment:
This is unique to IX maybe, but is it possible to only register one of these material flows based on a CONFIG argument for the IX model? The model only requires one of these for costing, so having all just adds bloat

Reply:
Not in a straightforward fashion, but of course anything can be done.

Right now the decorators ensure that the parameters are declared exactly once, even if two or more of the same unit is costed in a flowsheet.

The costing method doesn't need to use the decorators, however, to register the parameters. It could do it manually while ensuring they only get added once.